### ISSUE TITLE
Dynamic js imports have various issues, especially in scripts executed from the command line

### ISSUE THREAD

**Issue by dillydadally** [imba/imba#832]
### Description

Using the import() function in imba files to dynamically import modules has multiple issues.

If you run your script by typing "imba index.imba" into the command line, dynamic imports in your scripts seem to only work if the path passed to the import function is a hardcoded string. However, if you use a variable or string interpolation, it fails with the error:

`Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\Users\[username]\AppData\Local\Temp\tmp-48308-c3FnT6613Kqe\module.imba' imported from C:\Users\Dallin\AppData\Local\Temp\tmp-48308-c3FnT6613Kqe\index.imba
Did you mean to import ../../../../../../../projects/dynamic-import-test/module.imba?
    at __node_internal_captureLargerStackTrace (node:internal/errors:465:5)
    at finalizeResolution (node:internal/modules/esm/resolve:437:11)
    at moduleResolve (node:internal/modules/esm/resolve:1009:10)
    at defaultResolve (node:internal/modules/esm/resolve:1218:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:580:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:294:18)
    at ESMLoader.import (node:internal/modules/esm/loader:380:22)
    at importModuleDynamically (node:internal/modules/cjs/loader:1043:29)
    at importModuleDynamicallyWrapper (node:internal/vm/module:437:21)
    at importModuleDynamically (node:vm:381:46)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:35:14)
    at Object.<anonymous> (C:\Dropbox\projects\data imports\vicinity\run.js:2:1)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'ERR_MODULE_NOT_FOUND'
}`

When run via Vite or imba.new, it seems to work when passing a variable, but always produces errors if any string interpolation is used in the path argument. In certain situations, string interpolation should work with Vite but still produces a Vite error due to changes that Imba makes to the code before Vite analyzes the import statement.

The reproduction has more information about each of these issues and examples.

### Reproduction

https://scrimba.com/scrim/cg8E36UG

### System Environment

```text
System:
    OS: Windows 10 10.0.19045
    CPU: (24) x64 AMD Ryzen 9 3900XT 12-Core Processor
    Memory: 13.07 GB / 31.93 GB
  Binaries:
    Node: 16.15.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.18 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 9.1.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 111.0.5563.111
    Edge: Spartan (44.19041.1266.0), Chromium (111.0.1661.44)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    imba: ^2.0.0-alpha.229 => 2.0.0-alpha.229
```

**Comment by haikyuu** [comment]
Yes, this is a known issue. It happens because imba compiles template strings to string concatenation. And this breaks in Vite + rollup dynamic imports
